hw6.py

```diff
@@ -0,0 +1,98 @@
+class Exchanger(object):
+    _base = None
+
+    @staticmethod
+    def exchange(money, to_currency="USD"):
+        ratio = Exchanger._base.get((money.currency, to_currency))
+        if ratio is None:
+            # print("{} to {} ratio hasn't found! Used 1.0 as the ratio.")
+            ratio = 1.0
+        return Money(money.amount * ratio, to_currency)
+
+
+class ExchangerOnline(Exchanger):
+    @staticmethod
+    def exchange(money, to_currency="USD"):
+        if Exchanger._base is None:  # if dictionary is empty
+            data = ExchangerOnline.request_exchange_data()
+            Exchanger._base = ExchangerOnline.parse_exchange_data(data)
+        return Exchanger.exchange(money, to_currency)
+
+    @staticmethod
+    def request_exchange_data():
+        import json
+        import urllib.request
+        ak = "618af75c65d8fff43dda6794e6dee22e"
+        url = "http://www.apilayer.net/api/live?access_key={}".format(ak)
+        content = urllib.request.urlopen(url).read()
+        data = json.loads(content.decode('utf-8'))
+        return data
+
+    @staticmethod
+    def parse_exchange_data(data):
+        base = dict()
+        for currencies, rate in data['quotes'].items():
+            cur_from = currencies[0:3]
+            cur_to = currencies[3:6]
+            base.update({(cur_from, cur_to): rate})
```

Comment:
`base[(cur_from, cur_to)] = rate`.
Simple is better than complex (The Zen of Python)